lib/src/rive_render_box.dart

```diff
@@ -78,18 +154,21 @@ abstract class RiveRenderBox extends RenderBox {
 
   @override
   void attach(PipelineOwner owner) {
-    _stopwatch.start();
     super.attach(owner);
+    _stopwatch.start();
```

Comment:
Interesting, was this causing an issue?

Reply:
@luigi-rosso No, I do not think so. 

The reasoning behind this is as follows: the framework establishes paradigms for these life cycle methods and that is why it is in your best interest to follow them.  
For `State` it is documented, i.e. `initState` and `dispose` declare that `super.initState` should be called first thing in subclasses and `super.dispose` last thing. The same paradigm also goes for the `RenderObject` life cycle methods, even though it is not documented. You can see this in all `AbstractNode` subclasses in the framework.

-> in short, `super.attach` should be called first when overriding `attach`.

---

This can be because of assertions only (in the case of `initState`, which I [explained here](https://stackoverflow.com/a/58400324/6509751) or because some other logic is executed.  
In this specific case, it does not matter, however, this would be an easy source of error because you could `mixin` or extend a different class that executes logic in `attach` that should run before you start your `_stopwatch` or other code that you add later.

In the current setup, only the `PipelineOwner` is attached and the render object is marked as dirty in `super.attach`.

-> in the end, this change was only about following the paradigm. Sorry for the long-winded answer.

Reply:
@luigi-rosso I have to correct myself: it **is** actually document. So I could have spared the long explanation:

> Subclasses with children should override this method to **first call their inherited attach method**, and then attach all their children to the same owner.

([highlighting is mine](https://api.flutter.dev/flutter/foundation/AbstractNode/attach.html))